pkg/server/server.go

```diff
@@ -47,20 +48,20 @@ type Server interface {
 // The callbacks are invoked synchronously.
 type Callbacks interface {
 	// OnStreamOpen is called once an xDS stream is open with a stream ID and the type URL (or "" for ADS).
-	OnStreamOpen(int64, string)
+	OnStreamOpen(context.Context, int64, string) error
```

Comment:
Could do this in a backwards-compatible way by calling this `OnStreamOpenWithContext`, (and `OnFetchRequestWithContext` below). That said, this is a pretty easy change to take, so probably not worth it.

Reply:
Any change to the Callbacks interface would require an update. So even adding new functions like this would require implementations to add new no-op functions, which seems about the same amount of effort.